src/ui-test/utils/waitConditions.ts

```diff
@@ -77,7 +78,22 @@ export async function viewHasItem(content: ViewContent, section: string, item: s
     }
 }
 
-export async function updateFileText(oldText: string, newText: string): Promise<boolean> {
+export async function sectionHasItem(section: ViewSection, item: string, timePeriod = 2000): Promise<boolean> {
+    try {
+        const currentItem = await section.findItem(item);
+        if (currentItem !== undefined) {
+            return true;
+        } else {
+            await section.getDriver().sleep(timePeriod);
```

Comment:
what is the purpose of sleeping before returning false?

Reply:
This code was refactored from somewhere else, so I'm not sure what was the logic behind it.

Reply:
Can you find from where? In This PR, this code sounds to be new, i do not see any places from where it was deleted.

Reply:
from my understanding it will provide pause between next attempt of doing something.. so if we are waiting for something you can try every second or... try.. wait eg 3sec and try again

Reply:
this code is not called with several attempts.
if we want this behavior i think it should be handled directly inside this method and using a conditional wait

Reply:
Hi all,

That function used by function from utils:
https://github.com/camel-tooling/vscode-camelk/blob/38e4065ebe0c82c8d88b06359443393975953422/src/ui-test/utils/utils.ts#L125-L128

which is used in:

https://github.com/camel-tooling/vscode-camelk/blob/38e4065ebe0c82c8d88b06359443393975953422/src/ui-test/utils/utils.ts#L119-L123

https://github.com/camel-tooling/vscode-camelk/blob/38e4065ebe0c82c8d88b06359443393975953422/src/ui-test/utils/utils.ts#L130-L137

It can be replaced by an already tested function I think:

https://github.com/camel-tooling/vscode-camelk/blob/38e4065ebe0c82c8d88b06359443393975953422/src/ui-test/utils/waitConditions.ts#L71-L79

which is used here:
https://github.com/camel-tooling/vscode-camelk/blob/38e4065ebe0c82c8d88b06359443393975953422/src/ui-test/tests/dev_mode_test.ts#L96-L100

and here

https://github.com/camel-tooling/vscode-camelk/blob/38e4065ebe0c82c8d88b06359443393975953422/src/ui-test/tests/basic_mode_test.ts#L82-L86

Reply:
>  if we want this behavior i think it should be handled directly inside this method and using a conditional wait

I strongly believe we should refrain from having explicit code doing explicit things inside of tests and leaving them to more general setup tests and compose our tests from those general utils.

> @unsortedhashsets  said

I'll try changing it to the already tested function you mentioned, thanks!

Reply:
@unsortedhashsets I have been checking the functions, which underlay use

```
 export async function getCamelKIntegrationSection() {
    const section = await new SideBarView().getContent().getSection(uiTestConstants.extensionName) as CustomTreeSection;
    await section.expand();
    await VSBrowser.instance.driver.wait(async () => section.isExpanded());
    return section;
}
 ```
 
 As I had some failures because the section wasn't expanded so it couldn't find the items. From there, I went with the above function of "sectionHasItem" because it had the best interface for what we were using and sending the view content (sideBar) again seemed wasteful and misleading.
 
 I can work on refactoring the utils functions if we want but I have a feel that that might be outside of the scope of this PR.